caboose.go

```diff
@@ -55,23 +56,72 @@ type Config struct {
 	PoolLowWatermark int
 	// MaxRetrievalAttempts determines the number of times we will attempt to retrieve a block from the Saturn network before failing.
 	MaxRetrievalAttempts int
+
+	// MaxCidFailuresBeforeCoolDown is the maximum number of cid retrieval failures across the pool we will tolerate before we
+	// add the cid to the cool down cache.
+	MaxCidFailuresBeforeCoolDown int
+
+	// CidCoolDownDuration is duration of time a cid will stay in the cool down cache
+	// before we start making retrieval attempts for it.
+	CidCoolDownDuration time.Duration
+
+	// SaturnNodeCoolOff is the cool off duration for a saturn node once we determine that we shouldn't be sending requests to it for a while.
+	SaturnNodeCoolOff time.Duration
+
+	// MaxNCoolOff is the number of times we will cool off a node before downvoting it.
+	MaxNCoolOff int
 }
 
 const DefaultMaxRetries = 3
-const DefaultPoolFailureDownvoteDebounce = time.Second
-const DefaultPoolMembershipDebounce = 5 * time.Minute
+const DefaultPoolFailureDownvoteDebounce = 1 * time.Minute
+const DefaultPoolMembershipDebounce = 3 * DefaultPoolRefreshInterval
 const DefaultPoolLowWatermark = 5
 const DefaultSaturnRequestTimeout = 19 * time.Second
-const DefaultSaturnGlobalBlockFetchTimeout = 60 * time.Second
 const maxBlockSize = 4194305 // 4 Mib + 1 byte
 const DefaultOrchestratorEndpoint = "https://orchestrator.strn.pl/nodes/nearby?count=1000"
 const DefaultPoolRefreshInterval = 5 * time.Minute
 
+// we cool off sending requests to Saturn for a cid for a certain duration
+// if we've seen a certain number of failures for it already in a given duration.
+// NOTE: before getting creative here, make sure you dont break end user flow
+// described in https://github.com/ipni/storetheindex/pull/1344
+const DefaultMaxCidFailures = 3 * DefaultMaxRetries // this has to fail more than DefaultMaxRetries done for a single gateway request
+const DefaultCidCoolDownDuration = 1 * time.Minute  // how long will a sane person wait and stare at blank screen with "retry later" error before hitting F5?
```

Comment:
:information_source:  @aarshkshah1992 @willscott  I've changed these two values because `DefaultMaxCidFailures = 3` and `DefaultCidCoolDownDuration = 5m` were just breaking basic single user add&get described in https://github.com/ipni/storetheindex/pull/1344